lib/ansible/modules/clustering/kubevirt/k8s_service.py

```diff
@@ -2,64 +2,113 @@
-    selector:
-        description:
-            - Label selectors identify objects this Service should apply to.
-            - U(https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/)
-        required: true
-        type: dict
+description:
+  - Use Openshift Python SDK to manage Services on Kubernetes
 
 extends_documentation_fragment:
   - k8s_auth_options
-  - k8s_resource_options
+
+options:
+  resource_definition:
+    description:
+    - A partial YAML definition of the Service object being created/updated. Here you can define Kubernetes
+      Service Resource parameters not covered by this module's parameters.
+    - "NOTE: I(resource_definition) has lower priority than module parameters. If you try to define e.g.
+      I(metadata.namespace) here, that value will be ignored and I(metadata) used instead."
+    aliases:
+    - definition
+    - inline
+    type: dict
+  state:
+    description:
+    - Determines if an object should be created, patched, or deleted. When set to C(present), an object will be
+      created, if it does not already exist. If set to C(absent), an existing object will be deleted. If set to
+      C(present), an existing object will be patched, if its attributes differ from those specified using
+      module options and I(resource_definition).
+    default: present
+    choices:
+    - present
+    - absent
+  force:
+    description:
+    - If set to C(True), and I(state) is C(present), an existing object will be replaced.
+    default: false
+    type: bool
+  merge_type:
+    description:
+    - Whether to override the default patch merge approach with a specific type. By default, the strategic
+      merge will typically be used.
+    - For example, Custom Resource Definitions typically aren't updatable by the usual strategic merge. You may
+      want to use C(merge) if you see "strategic merge patch format is not supported"
+    - See U(https://kubernetes.io/docs/tasks/run-application/update-api-object-kubectl-patch/#use-a-json-merge-patch-to-update-a-deployment)
+    - Requires openshift >= 0.6.2
+    - If more than one merge_type is given, the merge_types will be tried in order
+    - If openshift >= 0.6.2, this defaults to C(['strategic-merge', 'merge']), which is ideal for using the same parameters
+      on resource kinds that combine Custom Resources and built-in resources. For openshift < 0.6.2, the default
+      is simply C(strategic-merge).
+    choices:
+    - json
+    - merge
+    - strategic-merge
+    type: list
+  name:
+    description:
+      - Use to specify a Service object name. If I(resource definition) is provided, the I(metadata.name)
+        value from the I(resource_definition) will override this option.
+    required: true
+    type: str
+  namespace:
+    description:
+      - Use to specify a Service object namespace. If I(resource definition) is provided, the
+        I(metadata.namespace) value from the I(resource_definition) will override this option.
```

Comment:
Isn't it the other way around?

Reply:
Good catch, you're right, I didn't correct this from older code.